sui_programmability/framework/sources/TestScenario.move

```diff
@@ -142,9 +149,33 @@ module Sui::TestScenario {
     /// Same as `remove_object`, but returns the object of type `T` with object ID `id`.
     /// Should only be used in cases where current tx sender has more than one object of
     /// type `T` in their inventory.
-    public fun remove_object_by_id<T: key>(_scenario: &mut Scenario, _id: ID): T {
-        // TODO: implement me
-        abort(100)
+    /// TODO: Perhaps also define can_remove_object_by_id? How to avoid code duplication?
+    public fun remove_object_by_id<T: key>(scenario: &mut Scenario, id: ID): T {
+        let sender = sender(scenario);
+        let objects: vector<T> = get_inventory<T>(
+            sender,
+            last_tx_start_index(scenario)
+        );
+        let object_opt = Option::none();
+        while (!Vector::is_empty(&objects)) {
+            let element = Vector::pop_back(&mut objects);
+            if (ID::id(&element) == &id) {
+                assert!(Option::is_none(&object_opt), EDUPLICATE_OBJCET_ID_FOUND);
+                Option::fill(&mut object_opt, element);
+            } else {
+                delete_object_for_testing(element);
```

Comment:
Previously from the naming I assumed this function will remove the object from the "global inventory" on the rust side, but it looks like this is just a [no-op](https://github.com/MystenLabs/sui/blob/317b91464ca631af4713f95ee8a1b79afb9e983f/sui_programmability/framework/src/natives/test_scenario.rs#L302). Maybe we can consider adding some comments  noting that it won't be removed from the inventory or give it another name like `drop_object_for_testing`